File: homework/hw2/code/game_movie.cpp
#include "game_movie.h"


game_movie::game_movie(char *name, int name_size,
                            int release_year, char *director, int director_size) : movie(name, name_size, release_year) {
    this->director = director;
    this->director_size = director_size;
}

game_movie::~game_movie() {
    delete[] name;
    delete[] director;
}

void game_movie::print(FILE *out) {
    fprintf(out, "[Game Movie]\n"
                 "-name: %s\n"
                 "-release year: %d\n"
                 "-director: %s\n"
                 "-function result: %lf\n\n", name, release_year, director, year_div_name());
}

File: homework/hw3/code/game_movie.py
from movie import Movie
import random
import string


class GameMovie(Movie):
    def __init__(self, *args):
        if len(args) > 0:
            super(GameMovie, self).__init__(*args[:-1])
            self._director_name = args[-1]
        else:
            super(GameMovie, self).__init__()
            self._director_name = ''.join(random.choice(string.ascii_lowercase)
                                          for _ in range(random.randint(1, 41)))

    def __str__(self):
        return """[Game Movie]:
              -name: {0}
              -release year: {1}
              -director name: {2}
              -function result: {3}\n""".format(self._name, self._release_year,
                                              self._director_name, self._release_year_div_name())
